Packet++/src/GtpLayer.cpp

```diff
@@ -0,0 +1,82 @@
+#define LOG_MODULE PacketLogModuleGtpLayer
+
+#include "GtpLayer.h"
+#include "UdpLayer.h"
+#include "IpUtils.h"
+#include "PayloadLayer.h"
+#include "DnsLayer.h"
+#include "DhcpLayer.h"
+#include "VxlanLayer.h"
+#include "SipLayer.h"
+#include "Logger.h"
+#include <string.h>
+#include <sstream>
+
+namespace pcpp
+{
+
+GtpLayer::GtpLayer(uint8_t flags, uint8_t type, uint16_t length, uint32_t teid, uint32_t seq)
+{
+	bool hasTeid = (flags & 0x8) != 0;
+	if (hasTeid){
+		m_DataLen = sizeof(gtphdr_teid);
+		m_Data = new uint8_t[m_DataLen];
+		memset(m_Data, 0, m_DataLen);
+		gtphdr_teid* gtpHdr = (gtphdr_teid*)m_Data;
+		gtpHdr->flags = flags;
+		gtpHdr->type = type;
+		gtpHdr->length = htons(length);
+		gtpHdr->teid = htonl(teid);
+		gtpHdr->seq = htonl(seq);
+	}else{
+		m_DataLen = sizeof(gtphdr);
+		m_Data = new uint8_t[m_DataLen];
+		memset(m_Data, 0, m_DataLen);
+		gtphdr* gtpHdr = (gtphdr*)m_Data;
+		gtpHdr->flags = flags;
+		gtpHdr->type = type;
+		gtpHdr->length = htons(length);
+		//gtpHdr->teid = htonl(teid);

```

Comment:
IS this correct?

Reply:
teid is not there if the teid flag is not set. I just commented that part in the version that doesn't have teid. But may be I should remove it and add a comment. Which one is better?